juju/model.py

```diff
@@ -1072,8 +1072,6 @@ def _get_series(self, entity_url, entity):
                     application_name = entity['Meta']['charm-metadata']['Name']
                 if not series:
                     series = self._get_series(entity_url, entity)
-                if not channel:
-                    channel = 'stable'
```

Comment:
Because it's wrong.  ;)  If an explicit rev is given, it may not be the stable channel.  There's also no actual need to provide a value for the channel, as we already resolve the charm to an explicit rev by querying the charm store.